lib/internal/Magento/Framework/App/Action/Action.php

```diff
@@ -17,14 +25,15 @@
  *
  * TODO: Remove this class. Allow implementation of Action Controllers by just implementing Action Interface.
  *
+ * phpcs:disable Magento2.Classes.AbstractApi
  * @api
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  * @SuppressWarnings(PHPMD.NumberOfChildren)
  */
 abstract class Action extends AbstractAction
```

Comment:
I believe we need to mark it as deprecated.

Reply:
It's API

Reply:
I believe we can depreacate even api 👍

Reply:
If you say so... :+1:

Reply:
After adding `@deprecated` it's failing Unit Tests :rofl: